invenio_accounts/views/rest.py

```diff
@@ -0,0 +1,533 @@
+    }
+
+    def success_response(self, user):
+        """Return a successful login response."""
+        return jsonify(default_user_payload(user))
+
+    def verify_login(self, user, password=None, **kwargs):
+        """Verify the login via password."""
+        if not user.password:
+            _abort(get_message('PASSWORD_NOT_SET')[0], 'password')
+        if not verify_and_update_password(password, user):
+            _abort(get_message('INVALID_PASSWORD')[0], 'password')
+        if requires_confirmation(user):
+            _abort(get_message('CONFIRMATION_REQUIRED')[0])
+        if not user.is_active:
+            _abort(get_message('DISABLED_ACCOUNT')[0])
+
+    def get_user(self, email=None, **kwargs):
+        """Retrieve a user by the provided arguments."""
+        return current_datastore.get_user(email)
+
+    def login_user(self, user):
+        """Perform any login actions."""
+        return login_user(user)
+
+    @use_kwargs(post_args)
+    def post(self, **kwargs):
+        """Verify and login a user."""
+        user = self.get_user(**kwargs)
+        self.verify_login(user, **kwargs)
+        self.login_user(user)
+        return self.success_response(user)
+
+
+class UserInfoView(MethodView):
+    """View to fetch info from current user."""
+
+    decorators = [login_required]
+
+    def success_response(self, user):
+        """Return a successful user info response."""
+        return jsonify(default_user_payload(user))
+
+    def get(self):
+        """Return user info."""
+        return self.success_response(current_user)
+
+
+class LogoutView(MethodView):
+    """View to logout a user."""
+
+    def logout_user(self):
+        """Perform any logout actions."""
+        if current_user.is_authenticated:
+            logout_user()
+
+    def success_response(self):
+        """Return a successful logout response."""
+        return jsonify({'message': 'User logged out.'})
+
+    def post(self):
+        """Logout a user."""
+        self.logout_user()
+        return self.success_response()
+
+
+class RegisterView(MethodView):
+    """View to register a new user."""
+
+    decorators = [user_already_authenticated]
+
+    post_args = {
+        'email': fields.Email(required=True, validate=[unique_user_email]),
+        'password': fields.String(
+            required=True, validate=[validate.Length(min=6, max=128)])
+    }
+
+    def login_user(self, user):
+        """Perform any login actions."""
+        if not current_security.confirmable or \
+                current_security.login_without_confirmation:
+            after_this_request(_commit)
+            login_user(user)
+
+    def success_response(self, user):
+        """Return a successful register response."""
+        return jsonify(default_user_payload(user))
+
+    @use_kwargs(post_args)
+    def post(self, **kwargs):
+        """Register a user."""
+        user = register_user(**kwargs)
```

Comment:
The UI view has a way to extend the form so you can ask for more fields. This doesn't seem to be supported? If not, then at least an issue should be opened it.

Reply:
Do you mean that there should be more args [here](https://github.com/inveniosoftware/invenio-accounts/pull/302/files#diff-223576a60bd9ea79383ec3cf2d503ebbR274)?

Reply:
Issues created:
-  https://github.com/inveniosoftware/invenio-accounts/issues/304
- https://github.com/inveniosoftware/invenio-userprofiles/issues/97